server/echoserver_test.go

```diff
@@ -0,0 +1,26 @@
+package main
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+)
+
+func TestEcho(t *testing.T) {
+	message := "hello test"
+	reader := strings.NewReader(message)
+	req, err := http.NewRequest("POST", "/echo", reader)
+	if err != nil {
+		t.Fatal(err)
+	}
+	rr := httptest.NewRecorder()
+	handler := http.HandlerFunc(echo)
+	handler.ServeHTTP(rr, req)
```

Comment:
What does wrapping `echo` in a handler get you over calling `echo` directly in the test code?

Reply:
There is no benefit in wrapping echo in a handler. I will call echo directly